packages/bpk-component-graphic-promotion/src/BpkGraphicPromo.tsx

```diff
@@ -0,0 +1,197 @@
+  Enter: 13 /* Enter */,
+  Space: 32 /* Space */,
+};
+
+const isAccessibilityClick = (event: React.KeyboardEvent<HTMLElement>) =>
+  Object.keys(ACCESSIBILITY_KEYS).includes(event.key) ||
+  Object.values(ACCESSIBILITY_KEYS).includes(event.keyCode || event.which);
+
+export const TEXT_ALIGN = {
+  start: 'start',
+  center: 'center',
+  end: 'end',
+} as const;
+
+export type Props = {
+  buttonText: string;
+  className?: string | null;
+  contentId?: string | null;
+  headline: string;
+  invertVertically: boolean;
+  onClick: () => void;
+  sponsor?: {
+    label: string;
+    logo: string;
+    altText: string;
+  } | null;
+  style?: {};
+  subheading?: string | null;
+  tagline?: string | null;
+  textAlign: typeof TEXT_ALIGN[keyof typeof TEXT_ALIGN];
+};
+
+const constructAriaLabel = (
+  buttonText: Props['buttonText'],
+  headline: Props['headline'],
+  sponsor: Props['sponsor'],
+  subheading: Props['subheading'],
+  tagline: Props['tagline'],
+) => {
+  const text: string[] = [];
+  const addText = (value: string) => value && text.push(value);
+
+  if (sponsor) {
+    addText(sponsor.label);
+    addText(sponsor.altText);
+  } else if (tagline) {
+    addText(tagline);
+  }
+
+  if (subheading) {
+    addText(subheading);
+  }
+
+  addText(headline);
+  addText(buttonText);
+
+  return text.join('. ');
+};
+
+const BpkGraphicPromo = ({
+  buttonText,
+  className = null,
+  contentId,
+  headline,
+  invertVertically = false,
+  onClick,
+  sponsor = null,
+  style = {},
+  subheading = null,
+  tagline = null,
+  textAlign,
+}: Props) => {
+  // FIXME: Use useCallback() here when React is updated.
+  const onClickWrapper = (event: React.MouseEvent<HTMLElement>) => {
+    event.stopPropagation();
+    onClick();
+  };
+  const onKeyWrapper = (event: React.KeyboardEvent<HTMLElement>) =>
+    isAccessibilityClick(event) && onClick();
+
+  const cardClasses = getClassName('bpk-card', 'bpk-graphic-promo', className);
+  const containerClasses = getClassName(
+    'bpk-graphic-promo__container',
+    `bpk-graphic-promo__container--${textAlign}`,
+    invertVertically && 'bpk-graphic-promo__container--inverted',
+  );
+
+  const getTextClasses = (baseClass: string) =>
+    getClassName(baseClass, `${baseClass}--${textAlign}`);
+
+  return (
+    // The card appears as a single component for the screen reader; its children are hidden. The card handles mouse
+    // clicks and key presses (Enter/Space) for the whole component, as described here:
+    // https://developer.mozilla.org/en-US/docs/Web/Accessibility/Keyboard-navigable_JavaScript_widgets
+    <div
+      className={cardClasses}
+      style={style}
+      role="link"
+      aria-label={constructAriaLabel(
+        buttonText,
+        headline,
+        sponsor,
+        subheading,
+        tagline,
+      )}
+      tabIndex={0}
+      onClick={onClickWrapper}
+      onKeyDown={onKeyWrapper}
+    >
+      <div
+        id={(contentId && contentId) || ''}
```

Comment:
`contentId` is nullable, add a conditional.

Reply:
Can this not be just and actually it runs as an if statement if its available or otherwise fallback to the `''`?
 
```suggestion
        id={contentId || ''}
```

Reply:
https://github.com/Skyscanner/backpack/pull/2535/commits/0bf909af62c69f32d0b708b4f479ea2b44a9f16b